src/explore-education-statistics-common/src/components/__tests__/Pagination.test.tsx

```diff
@@ -0,0 +1,270 @@
+import Pagination from '@common/components/Pagination';
+import { render as baseRender, screen, within } from '@testing-library/react';
+import React from 'react';
+
+describe('Pagination', () => {
+  test('renders the pagination without spacers when there are fewer than the max page numbers', () => {
+    render({ currentPage: 3, totalPages: 5 });
+
+    expect(
+      screen.getByRole('navigation', { name: 'Pagination navigation' }),
+    ).toBeInTheDocument();
+
+    const links = screen.getAllByRole('link');
+    expect(links).toHaveLength(7);
+
+    const listItems = screen.getAllByRole('listitem');
+    expect(listItems).toHaveLength(5);
+
+    // Previous link
+    expect(links[0]).toHaveAttribute('rel', 'prev');
+    expect(links[0]).toHaveAttribute('href', `/test-url?page=${2}`);
+    expect(within(links[0]).getByText('Previous')).toBeInTheDocument();
```

Comment:
Don't have to, but rather than using nested selectors like this, could just do:

```ts
expect(links[0]).toHaveTextContent('Previous');
```

Similar thing would need to be done for all other assertions throughout this test suite.